packages/elements/src/elementType.js

```diff
@@ -14,11 +14,11 @@
  * limitations under the License.
  */
 
-/**
- * External dependencies
- */
-import { __ } from '@googleforcreators/i18n';
+// @todo Create a custom hook to manage state.
+const elementTypes = [];
```

Comment:
@swissspidy Need your opinion here for handling `elementTypes` registration. Though a simple array works here however it doesn't feel right to keep it like this. I would expect it to become some state perhaps by keeping it in a custom hook, something like this

```js
/**
 * External dependencies.
 */
import { useState } from '@googleforcreators/react';

function useElementTypes() {
  const [ elementTypes, setElementTypes ] = useState( [] );

  const registerElementType = (elementType) => {
    setElementTypes( [ ...elementTypes, elementType ] );
  }

  return { elementTypes, registerElementType };
}

function useGetDefinitionForType( type ) {
  const { elementTypes } = useElementTypes();

  return () => elementTypes.find((el) => el.type === type);
}

export { useElementTypes, useGetDefinitionForType };
````

however, the problem is that if `elementTypes` becomes a state, it will have a cascading effect on other parts of the code using it for example `getDefinitionForType` , `createNewElement` would have to become custom hooks too. Then a lot of areas where these utility functions are being used in `story-editor` will have to be refactored to handle hooks.

Ideally, I would wish to keep them as static functions ( for example `registerblocktype` in Gutenberg ) but unfortunately, we do not have a central store that can do that ( like data package in Gutenberg ).

Reply:
I see the problem and I am on the same page as you. Using a hook here would indeed cause quite a cascading effect. So to me a central/global store also sounds more appealing right now.

This is a good reason/opportunity to revisit this in a dedicated ticket explaining the use case.

Reply:
Yes and like you previously asked, that central store may help also us in making data from the story editor available to plugins like Yoast SEO.

Reply:
@swissspidy I suppose #10614 discussion and the actual work might take some time and `story-editor` is constantly getting updated. I am wondering what to do with this PR as it would be a little difficult to keep it in sync and my other package organization tickets may also depend on this PR.

Reply:
As discussed on Slack, using something like Redux or zustand can be done in follow-up tickets. This here works as a temporary solution in the meantime.

Reply:
Curious about the proposed use case here. So the idea is that once the app is running a plugin might add a new element type that necessitate a re-render in the application?

Or would this be some data we just need at build time?

Reply:
> So the idea is that once the app is running a plugin might add a new element type that necessitates a re-render in the application?

That's the wish when we set up a proper data package.

Reply:
Cool. I'm excited to see what you whip up! should definitely be easier to discuss/visualize once we have something a little more tangible to look at.